File: 48-rotate-image/48-rotate-image.py
class Solution:
    def rotate(self, matrix: List[List[int]]) -> None:
        """
        Do not return anything, modify matrix in-place instead.
        """
        i = 0
        j = len(matrix)-1
        while(i < j):
            for k in range(len(matrix[0])):
                matrix[i][k], matrix[j][k] = matrix[j][k], matrix[i][k]
            i += 1
            j -= 1
        i = 0
        while i < len(matrix):
            for k in range(i+1, len(matrix[0])):
                matrix[i][k], matrix[k][i] = matrix[k][i], matrix[i][k]
            i += 1
        
        
        

File: 48-rotate-image/48-rotate-image.java
class Solution {
    public void rotate(int[][] matrix) {
        int m = matrix.length;
        int n = matrix[0].length;
        int i=0, j=m-1;
        while(i < j){
            for(int k=0;k<n;k++){
                int temp = matrix[i][k];
                matrix[i][k] = matrix[j][k];
                matrix[j][k] = temp;
            }
            i++;
            j--;
        }
        i = 0;
        while(i < m){
            for(j=i+1;j<n;j++){
                int temp = matrix[i][j];
                matrix[i][j] = matrix[j][i];
                matrix[j][i] = temp;
            }
            i++;
        }
        
    }
}